mantis-server/mantis-server-agent/src/main/java/io/mantisrx/server/agent/TaskExecutorStarter.java

```diff
@@ -68,6 +69,7 @@ protected void shutDown() throws Exception {
             .exceptionally(throwable -> null)
             .thenCompose(dontCare -> Services.stopAsync(highAvailabilityServices,
                 MoreExecutors.directExecutor()))
+            .thenRunAsync(rpcSystem::close)
```

Comment:
Instead of doing this in TaskExecutorStarter, why not do this from the TaskExecutor itself?

Reply:
+1

Reply:
we pass `RpcService` in `TaskExecutor` which doesn't have any reference to `RpcSystem`. I think this is the right way to close our custom rpc system.

Reply:
does this gets run if the worker crashed?

Reply:
Yes it does.